src/main/java/org/eeichinger/servicevirtualisation/jdbc/MockResultSetHelper.java

```diff
@@ -105,15 +109,21 @@ public List<Object> toRowValues() {
         }
     }
 
-    private static String getNilableElementText(Element col) {
+    protected Object getNilableElementText(Element col) {
         String val = col.getText();
         if ("true".equalsIgnoreCase(col.getAttributeValue("nil", nsXsi))) {
-            val = null;
+            return null;
+        }
+        else {
+            String type = col.getAttributeValue("type", nsXsi);
+            if (type != null) {
```

Comment:
This behaviour is on by default, which seems safe since it will only happen if xs:type is in the xml.